base_layer/p2p/src/services/liveness/state.rs

```diff
@@ -192,15 +192,30 @@ impl LivenessState {
         self.peer_latency.get(node_id).map(|latency| latency.calc_average())
     }
 
+    /// Add a NodeId to be monitored. If the NodeID already exists then increment the count of how many requests for
+    /// monitoring have occured.
     pub fn add_node_id(&mut self, node_id: &NodeId) {
-        if self.nodes_to_monitor.contains_key(node_id) {
-            return;
+        if let Some(n) = self.nodes_to_monitor.get_mut(node_id) {
+            n.number_of_monitor_requests += 1;
+        } else {
+            let _ = self.nodes_to_monitor.insert(node_id.clone(), NodeStats::new());
         }
-        let _ = self.nodes_to_monitor.insert(node_id.clone(), NodeStats::new());
     }
 
+    /// Reduce the count of how many request has been received to monitor NodeId, if this would reduce the count to zero
```

Comment:
Not a fan of this, feels like unexpected behaviour to call "remove_node_id" and not have the node id removed. But should be ok for now as the transaction service is the only component that uses monitored nodes AFAIK